### ISSUE TITLE
Implement ActivityPub in order to provide an update feed for the Fediverse

### ISSUE THREAD

**Issue by novoid** [novoid/lazyblorg#71]
I would like to explore the possibility to generate a feed in the form of ActivityPub so that users in the [Fediverse](https://en.wikipedia.org/wiki/Fediverse) are able to follow blog article updates.

I don't know ActivityPub at all. I would assume that it's a similar format to Atom (which is already implemented): a static text page that contains the most recent articles.

- [ ] confirm or falsify assumptions above
- [ ] read https://socialhub.activitypub.rocks/t/guide-for-new-activitypub-implementers/479

Somebody was using ActivityPub to follow on Mastodon and retooting his own articles: https://www.zylstra.org/blog/2018/10/activity-pub-stream-from-rss/ -> this is just an example with a screenshot how it could look like on Mastodon.

**Comment by xeruf** [comment]
https://writefreely.org implements a minimal blogging system that federates, maybe worth looking into as a foundation?

**Comment by xeruf** [comment]
> I would assume that it's a similar format to Atom (which is already implemented): a static text page that contains the most recent articles.

Nope, you need a server running that responds to follow requests and actively notifies all subscribers on a new post.
So this will have to work a bit differently than the current deployment.

But I would highly appreciate this feature, it was one of the original issues holding me back from using lazyblorg to setup my blog!
Maybe it would also be better to not implement ActivityPub at all and instead add the ability to call a webhook upon updating the blog, which can then be handled by another AP server somehow...

**Comment by novoid** [comment]
> > I would assume that it's a similar format to Atom (which is already implemented): a static text page that contains the most recent articles.
> 
> Nope, you need a server running that responds to follow requests and actively notifies all subscribers on a new post. So this will have to work a bit differently than the current deployment.

Oh, this dramatically reduces the chance of an ActivityPub implementation with lazyblorg as it seems to add substantial level of complexity and implementation effort.

I'm open for contributions though. They need to be **implemented as an optional feature** since one of the basic premises of lazyblorg is that it is a static web site generator which doesn't require anything else on the server.

**Comment by novoid** [comment]
Kev has some really good thoughts about **not integrating** a personal blog to services like ActivityPub: https://kevquirk.com/does-a-blog-need-to-integrate/

At the moment, I'm inclined to agree here. I don't know what the gain is, the amount of effort seems substantial (including introducing active elements to my static site) and what are my arguments to stop after integrating with only one standard. So it's a rabbit hole I might enter here ...

**Comment by snarfed** [comment]
Feel free to try https://fed.brid.gy/ ! It adds ActivityPub to web sites and blogs as a service.

**Comment by xeruf** [comment]
@novoid the big plus of focusing on this standard is that it could handle comments as well :)

**Comment by xeruf** [comment]
There is a new, arguably better protocol rising up: https://nostr.com/
Just send a bit of JSON to a relay and you are done.
Could probably also handle comments #40
